cmd/kubelet/app/server.go

```diff
@@ -841,15 +841,26 @@ func buildKubeletClientConfig(ctx context.Context, s *options.KubeletServer, nod
 		// we set exitAfter to five minutes because we use this client configuration to request new certs - if we are unable
 		// to request new certs, we will be unable to continue normal operation. Exiting the process allows a wrapper
 		// or the bootstrapping credentials to potentially lay down new initial config.
-		closeAllConns, err := kubeletcertificate.UpdateTransport(wait.NeverStop, transportConfig, clientCertificateManager, 5*time.Minute)
```

Comment:
I think it would be clearer to keep this var `closeAllConns` (since that's actually what this one does), and set up a separate `var onHeartbeatFailure func()` below which we set either to `closeAllConns` or to `func() { utilnet.CloseIdleConnectionsFor(transportConfig.Transport) }`

Reply:
yep, that is right, let me fix it